src/main3.cpp

```diff
@@ -94,23 +100,51 @@ int main(int argc, char** argv) {
         return 1;
     }
 
-    fprintf(stdout, "Server ready for inference\n");
-
-    // infer 1
-    // infer 2
-    // infer 3
-
-    // Application loop if required (C++):
-    // while (shutdown_request == 0) {
-    //     std::this_thread::sleep_for(std::chrono::milliseconds(200));
-    // }
-
-    fprintf(stdout, "No more job to be done, will shut down\n");
+    std::cout << "Server ready for inference" << std::endl;
+
+    // prepare request
+    OVMS_InferenceRequest* request{nullptr};
+    res = OVMS_InferenceRequestNew(&request, MODEL_NAME, MODEL_VERSION);
+    OVMS_InferenceRequestAddInput(request, INPUT_NAME, OVMS_DATATYPE_FP32, SHAPE, DIM_COUNT);
+    std::array<float, SHAPE[1]> data{0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    OVMS_InferenceRequestInputSetData(request, INPUT_NAME, reinterpret_cast<void*>(data.data()), sizeof(float) * data.size(), OVMS_BUFFERTYPE_CPU, 0);
+
+    // run sync request
+    OVMS_InferenceResponse* response = nullptr;
+    OVMS_Inference(srv, request, &response);
+
+    // read output
+    uint32_t outputCount = 0;
+    OVMS_InferenceResponseGetOutputCount(response, &outputCount);
+    const void* voutputData;
+    size_t bytesize = 0;
+    uint32_t outputId = outputCount - 1;
+    OVMS_DataType datatype = (OVMS_DataType)42;
+    const uint64_t* shape{nullptr};
+    uint32_t dimCount = 0;
+    OVMS_BufferType bufferType = (OVMS_BufferType)42;
+    uint32_t deviceId = 42;
+    const char* outputName{nullptr};
+    OVMS_InferenceResponseGetOutput(response, outputId, &outputName, &datatype, &shape, &dimCount, &voutputData, &bytesize, &bufferType, &deviceId);
+
+    std::stringstream ss;
+    ss << "Got response from OVMS via C-API. "
+       << "Request for model: " << MODEL_NAME
+       << "; version: " << MODEL_VERSION
+       << "ms; output name: " << outputName
+       << "; response with values:\n";
+    for (size_t i = 0; i < shape[1]; ++i) {
+        ss << *(reinterpret_cast<const float*>(voutputData) + i) << " ";
+    }
+    // comment line below to have app running similarly to OVMS
+    shutdown_request = 1;
+    while (shutdown_request == 0) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(200));
```

Comment:
This needs to be removed if we want to keep in in Dockerfile as part of CI check.

Reply:
check L139-140. This is left intentionally to have easy way to keep gRPC alive and provide example

Reply:
Does CI currently pass? Running cpp test app is currently in dockerfile as functional test. We should remove it then